GitUI/BranchTreePanel/RepoObjectsTree.Nodes.cs

```diff
@@ -165,51 +174,45 @@ protected void ReloadNodes(Func<CancellationToken, Task> loadNodesTask)
 
                 ThreadHelper.JoinableTaskFactory.RunAsync(async () =>
                 {
+                    Nodes.Clear();
+
+                    var token = _reloadCancellationTokenSequence.Next();
```

Comment:
`_reloadCancellationTokenSequence` is shared by each root node reload action which is wrong and works only because there is no recheck after the async action if the token is valid (see the comment below).
Each invocation of ReloadNodes cancels the previous one. If I refresh Branches and then refresh Submodules the Branches refresh action will get cancelled.

Reply:
> `_reloadCancellationTokenSequence` is shared by each root node reload action which is wrong and works only because there is no recheck after the async action if the token is valid (see the comment above).
> Each invocation of ReloadNodes cancels the previous one. If I refresh Branches and then refresh Submodules the Branches refresh action will get cancelled.

As I explained above, this isn't shared across root nodes. There is one instance per Tree, so cancelling in one does not affect the others.

Reply:
Thank you for the thorough explanation. Yes you are right, I didn't notice the change you explained.